crates/core/tedge/src/main.rs

```diff
@@ -4,24 +4,27 @@
 use anyhow::Context;
 use clap::Parser;
 use tedge_users::UserManager;
-
 mod cli;
 mod command;
 mod error;
 mod system_services;
+use tedge_utils::file::create_directory_with_user_group;
 
 type ConfigError = crate::error::TEdgeError;
 
 use command::{BuildCommand, BuildContext};
 
 fn main() -> anyhow::Result<()> {
-    let user_manager = UserManager::new();
-
-    let _user_guard = user_manager.become_user(tedge_users::TEDGE_USER)?;
-
     let opt = cli::Opt::parse();
+    let user_manager = UserManager::new();
+    if opt.init {
+        println!("Initialize the tedge");
+        initialize_tedge()?;
+        return Ok(());
+    }
 
     let tedge_config_location = tedge_config::TEdgeConfigLocation::from_custom_root(opt.config_dir);
+    let _user_guard = user_manager.become_user(tedge_users::TEDGE_USER)?;
```

Comment:
The two steps `UserManager::new()` and `user_manager.become_user()` should not be put apart.

```suggestion
    let opt = cli::Opt::parse();
    if opt.init {
        println!("Initialize the tedge");
        initialize_tedge()?;
        return Ok(());
    }
    
    let user_manager = UserManager::new();
    let _user_guard = user_manager.become_user(tedge_users::TEDGE_USER)?;
        
    let tedge_config_location = tedge_config::TEdgeConfigLocation::from_custom_root(opt.config_dir);
    
```

Reply:
fixed

Reply:
I don't see the fixed.

Reply:
forgot to push the code. Now its pushed